Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -0,0 +1,479 @@
+#include "WarpXOpenPMD.H"
+#include "FieldIO.H"  // for getReversedVec
+
+WarpXOpenPMDPlot::WarpXOpenPMDPlot(bool oneFilePerTS,
+                       const std::string& openPMDFileType)
+  :m_Series(nullptr),
+   m_OneFilePerTS(oneFilePerTS),
+   m_OpenPMDFileType(openPMDFileType)
+{}
+
+WarpXOpenPMDPlot::~WarpXOpenPMDPlot()
+{
+  if (nullptr != m_Series) {
+    m_Series->flush();
+    delete m_Series;
+  }
+}
+
+
+//
+//
+//
+void WarpXOpenPMDPlot::GetFileName(std::string& filename, int ts)
+{
+  std::string dir = "diags/";
```

Comment:
I would suggest to simplify this logic in the following way:
```C++
    std::string const basename = amrex::Concatenate("diags/", m_OpenPMDFileType, "/simData");
    if (m_OneFilePerTS)
        filename = base_dir.append("_%07T");

    filename.append(".").append(m_OpenPMDFileType);
```

Reply:
Contatenate is not used that way, but I see your point.